tests/unit/test_llm.py

```diff
@@ -363,3 +364,88 @@ def test_llm_cloudflare_blockage(mock_litellm_completion, default_config):
 
     # Ensure the completion was called
     mock_litellm_completion.assert_called_once()
+
+
+@patch('openhands.llm.llm.litellm_completion')
+def test_token_count_with_usage_data(mock_litellm_completion, default_config):
+    # Mock a response with usage data
+    from litellm import Message as LiteLLMMessage
+    from litellm.types.utils import Choices, ModelResponse, Usage
+
+    mock_response = ModelResponse(
+        id='test-id',
+        choices=[
+            Choices(
+                message=LiteLLMMessage(role='assistant', content='Test response'),
+                finish_reason='stop',
+                index=0,
+            )
+        ],
+        created=1234567890,
+        model='test-model',
+        object='chat.completion',
+        usage=Usage(prompt_tokens=10, completion_tokens=20, total_tokens=30),
+    )
+    mock_litellm_completion.return_value = mock_response
+
+    llm = LLM(default_config)
+    response = llm.completion(
+        messages=[Message(role='user', content=[TextContent(text='Hello!')])]
+    )
+
+    # Verify the response has usage data
+    assert response.usage.prompt_tokens == 10
+    assert response.usage.completion_tokens == 20
+    assert response.usage.total_tokens == 30
+
+    # Verify get_token_count uses the stored token counts
+    messages = [
+        Message(role='user', content=[TextContent(text='Hello!')], total_tokens=30)
+    ]
```

Comment:
@openhands-agent Wait a minute. Look at this carefully. You must test here if the llm.py code did store "30" before, because you wrote it already in Usage. Read above, it was already in Usage. You must not create a new Message and send it again. Then the test is not testing the right thing: that the LLM class read it from Usage. Remove this addition of another Message, and test what you already created above.